packages/bentley/grouping-mapping-widget/src/api/reportingClient.ts

```diff
@@ -0,0 +1,385 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { IModelApp } from "@bentley/imodeljs-frontend";
+import { AuthorizationClient } from "@bentley/itwin-client";
+import { getConfig } from "../config";
+
+import { BASE_PATH, CalculatedPropertyCreateReportingAPI, CalculatedPropertyUpdateReportingAPI, GroupCreateReportingAPI, GroupPropertyCreateReportingAPI, GroupPropertyUpdateReportingAPI, GroupUpdateReportingAPI, MappingCollectionReportingAPI, MappingCopyReportingAPI, MappingCreateReportingAPI, MappingReportingAPI, MappingsApi, MappingUpdateReportingAPI } from "./generated/api";
+
+const ACCEPT = "application/vnd.bentley.itwin-platform.v1+json";
+
+export const getUrlPrefix = () => {
+  const config = getConfig();
+  switch (config.buddi?.region) {
+    case "101":
+    case "103":
+      return "dev";
+    case "102":
+      return "qa";
+    default:
+      return "";
+  }
+};
+
+const prefixUrl = (baseUrl?: string, prefix?: string) => {
+  if (prefix && baseUrl) {
+    return baseUrl.replace("api.bentley.com", `${prefix}-api.bentley.com`);
+  }
+  return baseUrl;
+};
+
+// To be only used within Viewer
+class ReportingClient {
+  private _mappingsApi: MappingsApi;
+  constructor() {
+    const baseUrl = prefixUrl(BASE_PATH, getUrlPrefix());
+    this._mappingsApi = new MappingsApi(undefined, baseUrl);
+  }
+
+  public async getMappings(iModelId: string) {
+    const mappings: Array<MappingReportingAPI> = [];
+
+    const _accessToken = await (
+      IModelApp.authorizationClient as AuthorizationClient
```

Comment:
did you need to cast it here?

Reply:
It can be potentially undefined, which to be honest is impossible due to how the widget needs the viewer to be loaded.

Reply:
if its undefined then casting it here wont really help, casting it here just obfuscates the issue, and can throw an uncaught exception the way it is

i think this is more preferrable
const _accessToken = await IModelApp.authorizationClient?.getAccessToken() : "";

Reply:
Sounds good. Will change to this.